sdk/tests/conformance2/reading/read-pixels-into-pixel-pack-buffer.html

```diff
@@ -0,0 +1,269 @@
+** copy of this software and/or associated documentation files (the
+** "Materials"), to deal in the Materials without restriction, including
+** without limitation the rights to use, copy, modify, merge, publish,
+** distribute, sublicense, and/or sell copies of the Materials, and to
+** permit persons to whom the Materials are furnished to do so, subject to
+** the following conditions:
+**
+** The above copyright notice and this permission notice shall be included
+** in all copies or substantial portions of the Materials.
+**
+** THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+-->
+
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>WebGL 2 Conformance Test: ReadPixels Into Pixel Pack Buffer.</title>
+<link rel="stylesheet" href="../../resources/js-test-style.css"/>
+<script src="../../js/js-test-pre.js"></script>
+<script src="../../js/webgl-test-utils.js"></script>
+</head>
+<body>
+<canvas id="example" width="2" height="2"></canvas>
+<div id="description"></div>
+<div id="console"></div>
+<script>
+"use strict";
+
+function checkFormatAndType()
+{
+    debug("");
+    debug("check format / type");
+    var validFormat = [gl.LUMINANCE, gl.RED, gl.RG_INTEGER, gl.RGBA_INTEGER];
+    var validType = [gl.UNSIGNED_SHORT, gl.BYTE, gl.INT, gl.UNSIGNED_INT_2_10_10_10_REV];
+    for (var ff = 0; ff < validFormat.length; ++ff) {
+        var format = validFormat[ff];
+        gl.readPixels(0, 0, 1, 1, format, gl.UNSIGNED_BYTE, 0);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "Format should be able to read as " + wtu.glEnumToString(gl, format));
+    }
+    for (var tt = 0; tt < validType.length; ++tt) {
+        var type = validType[tt];
+        gl.readPixels(0, 0, 1, 1, gl.RGBA, type, 0);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "Type should be able to read as " + wtu.glEnumToString(gl, type));
+    }
+
+    var invalidFormat = [gl.DEPTH_COMPONENT, gl.DEPTH_STENCIL, gl.R8, gl.RGBA4];
+    var invalidType = [gl.UNSIGNED_INT_24_8];
+    for (var ff = 0; ff < invalidFormat.length; ++ff) {
+        var format = invalidFormat[ff];
+        gl.readPixels(0, 0, 1, 1, format, gl.UNSIGNED_BYTE, 0);
+        wtu.glErrorShouldBe(gl, gl.INVALID_ENUM, "Format should not be able to read as " + wtu.glEnumToString(gl, format));
+    }
+    for (var tt = 0; tt < invalidType.length; ++tt) {
+        var type = invalidType[tt];
+        gl.readPixels(0, 0, 1, 1, gl.RGBA, type, 0);
+        wtu.glErrorShouldBe(gl, gl.INVALID_ENUM, "Type should not be able to read as " + wtu.glEnumToString(gl, type));
+    }
```

Comment:
This is incorrect.  ReadPixels only supports one basic format/type, one implementation specific format/type.  Only with RGB10_A2, a third format/type is allowed.  Your test is saying all these formats and types have to be allowed.